accessors/spanner/spanner_accessor.go

```diff
@@ -201,3 +233,211 @@ func (sp *SpannerAccessorImpl) CreateChangeStream(ctx context.Context, adminClie
+		// a) backticks around the database name, and
+		// b) DDL statements as part of a CreateDatabase operation (so schema
+		// must be set using a separate UpdateDatabase operation).
+		req.CreateStatement = "CREATE DATABASE \"" + dbName + "\""
+		req.DatabaseDialect = adminpb.DatabaseDialect_POSTGRESQL
+	} else {
+		req.CreateStatement = "CREATE DATABASE `" + dbName + "`"
+		if migrationType == constants.DATAFLOW_MIGRATION {
+			req.ExtraStatements = conv.SpSchema.GetDDL(ddl.Config{Comments: false, ProtectIds: true, Tables: true, ForeignKeys: true, SpDialect: conv.SpDialect, Source: driver})
+		} else {
+			req.ExtraStatements = conv.SpSchema.GetDDL(ddl.Config{Comments: false, ProtectIds: true, Tables: true, ForeignKeys: false, SpDialect: conv.SpDialect, Source: driver})
+		}
+
+	}
+
+	op, err := adminClient.CreateDatabase(ctx, req)
+	if err != nil {
+		return fmt.Errorf("can't build CreateDatabaseRequest: %w", utils.AnalyzeError(err, dbURI))
+	}
+	if _, err := op.Wait(ctx); err != nil {
+		return fmt.Errorf("createDatabase call failed: %w", utils.AnalyzeError(err, dbURI))
+	}
+
+	if conv.SpDialect == constants.DIALECT_POSTGRESQL {
+		// Update schema separately for PG databases.
+		return sp.UpdateDatabase(ctx, adminClient, dbURI, conv, driver)
+	}
+	return nil
+}
+
+// UpdateDatabase updates an existing spanner database.
+func (sp *SpannerAccessorImpl) UpdateDatabase(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI string, conv *internal.Conv, driver string) error {
+	// The schema we send to Spanner excludes comments (since Cloud
+	// Spanner DDL doesn't accept them), and protects table and col names
+	// using backticks (to avoid any issues with Spanner reserved words).
+	// Foreign Keys are set to false since we create them post data migration.
+	schema := conv.SpSchema.GetDDL(ddl.Config{Comments: false, ProtectIds: true, Tables: true, ForeignKeys: false, SpDialect: conv.SpDialect, Source: driver})
+	req := &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dbURI,
+		Statements: schema,
+	}
+	// Update queries for postgres as target db return response after more
+	// than 1 min for large schemas, therefore, timeout is specified as 5 minutes
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+	op, err := adminClient.UpdateDatabaseDdl(ctx, req)
+	if err != nil {
+		return fmt.Errorf("can't build UpdateDatabaseDdlRequest: %w", utils.AnalyzeError(err, dbURI))
+	}
+	if err := op.Wait(ctx); err != nil {
+		return fmt.Errorf("UpdateDatabaseDdl call failed: %w", utils.AnalyzeError(err, dbURI))
+	}
+	return nil
+}
+
+// CreatesOrUpdatesDatabase updates an existing Spanner database or creates a new one if one does not exist.
+func (sp *SpannerAccessorImpl) CreateOrUpdateDatabase(ctx context.Context, adminClient spanneradmin.AdminClient, dbURI, driver string, conv *internal.Conv, migrationType string) error {
+	dbExists, err := sp.VerifyDb(ctx, adminClient, dbURI)
+	if err != nil {
+		return err
+	}
```

Comment:
There seem to be a bunch  of missing coverage mainly on the error flows in this file. We should try to cover these as well.